mountpoint-s3-fs/src/memory/pages.rs

```diff
@@ -111,18 +122,37 @@ impl Page {
     /// Returns whether the page was invalidated. If `true` the page is ready to be
     /// removed from the pool.
     pub fn invalidate_if_empty(&self) -> bool {
-        let mut bitmask = self.inner.free_bitmask.lock().unwrap();
+        let mut bitmask = self.inner.reserved_bitmask.lock().unwrap();
         if *bitmask != 0 {
             return false;
         }
-        // Prevent further use.
+        self.inner.stats.remove_empty_page();
+        // Mark the page as full, even if no buffer is currently reserved. If a new request
```

Comment:
*even though no buffer is currently reserved. We just checked this, and the bitmask is still locked, so we know it can't have changed